skeleton/template.yaml

```diff
@@ -80,12 +80,36 @@ spec:
                   title: Model Server Endpoint
                   type: string
                   description: "The endpoint for an existing model server."
+                includeModelEndpointSecret:
+                  title: Bearer Authentication Required?
+                  type: boolean
+                  default: false
                 # SED_LLM_SERVER_START
                 modelName:
                   title: Model Name
                   type: string
                   ui:help: "The name of the model deployed on the model server you would like to use."
                 # SED_LLM_SERVER_END
+              dependencies:
+                includeModelEndpointSecret:
+                  allOf:
+                    - if:
+                        properties:
+                          includeModelEndpointSecret:
+                            const: true
+                      then:
+                        properties:
+                          modelEndpointSecretName:
+                            title: Model Endpoint Secret Name
+                            ui:help: Provide the name of the Secret containing your bearer token.
```

Comment:
Have you tried this secret reference in a custom namespace? 🤔 

In the templates wizard,

1. We ask for the Namespace to be deployed
2. Deployment references the Secret in the same Namespace as Deployment https://github.com/redhat-ai-dev/ai-lab-template/pull/40/files#diff-f3f000f5975fd959a1fd26857e77e8fcc8c892cc11f8cd83bca93e5b030b7591R39 
3. Default Namespace is fine, but if user is providing a custom Namespace via Step 1; they will have to create the Namespace before creating the Secret?

Reply:
Yes, you need to have the secret created in your namespace upfront in order to provide it here.

> Default Namespace is fine, but if user is providing a custom Namespace via Step 1; they will have to create the Namespace before creating the Secret?

I think we don't create the default namespace (meaning the `ai-rhdh-app-development`) when we run the installer right? So for every case the namespace and the secret should be there beforehand. Otherwise I'm not sure how we can avoid passing the token itself in the wizzard. Might be better if we capture that in the `ui:help`?

Reply:
Currently `ai-rhdh-app-development`, `ai-rhdh-app-stage` and `ai-rhdh-app-prod` are created when the installer is run but we also added the Tekton tasks that allow you to enter any namespace in the template and it sets it up for you

Reply:
> Currently ai-rhdh-app-development, ai-rhdh-app-stage and ai-rhdh-app-prod are created when the installer is run but we also added the Tekton tasks that allow you to enter any namespace in the template and it sets it up for you

Yeah tbh as the installer is in progress I wasn't exactly sure how it's done. But in general I think the principal is the same. We need to define the secret prior to the template creation.

I agree with @maysunfaisal and @johnmcollier I'll try adding more context in the `ui:help field`.

Reply:
Tried to add more info here: [af11a9e](https://github.com/redhat-ai-dev/ai-lab-template/pull/40/commits/af11a9ea05ac40c13233f6d23faebc76b1784687)